bin/host/src/server.rs

```diff
@@ -20,6 +19,17 @@ pub struct PreimageServer<P, H, B> {
     backend: Arc<B>,
 }
 
+/// An error that can occur when handling preimage requests
+#[derive(Debug, thiserror::Error)]
+pub enum PreimageServerError {
```

Comment:
This will cause ci to error because the type is only visible within the `server` module. It has to be visible at the top-level of the crate so linting will pass.

If you run `just l` locally which runs linting, you'll see a visibility error.

The solution is in the host's `lib.rs` file, make the `PreimageServerError` visible by adding it to the `pub use server::` line [here](https://github.com/op-rs/kona/blob/main/bin/host/src/lib.rs#L5).

You'll also need to do this for the other error types introduced in this PR.